aries_cloudagent/did/indy/routes.py

```diff
@@ -0,0 +1,76 @@
+"""DID INDY routes."""
+
+from http import HTTPStatus
+
+from aiohttp import web
+from aiohttp_apispec import docs, request_schema, response_schema
+from marshmallow import fields
+
+from ...admin.decorators.auth import tenant_authentication
+from ...admin.request_context import AdminRequestContext
+from ...did.indy.indy_manager import DidIndyManager
+from ...messaging.models.openapi import OpenAPISchema
+from ...wallet.error import WalletError
+
+
+class CreateRequestSchema(OpenAPISchema):
+    """Parameters and validators for create DID endpoint."""
+
+    seed = fields.Str(
+        required=False,
+        metadata={
+            "description": (
+                "Optional seed to use for DID, Must be enabled in configuration before"
+                " use."
+            ),
+            "example": "000000000000000000000000Trustee1",
+        },
+    )
+
+
+class CreateResponseSchema(OpenAPISchema):
+    """Response schema for create DID endpoint."""
+
+    did = fields.Str(description="DID created", example="did:indy:DFZgMggBEXcZFVQ2ZBTwdr")
+    verkey = fields.Str(
+        description="Verification key",
+        example="BnSWTUQmdYCewSGFrRUhT6LmKdcCcSzRGqWXMPnEP168",
+    )
+
+
+@docs(tags=["did"], summary="Create an INDY DID")
+@request_schema(CreateRequestSchema())
+@response_schema(CreateResponseSchema, HTTPStatus.OK)
+@tenant_authentication
+async def create_indy_did(request: web.BaseRequest):
+    """Create a INDY DID."""
+    context: AdminRequestContext = request["context"]
+    body = await request.json()
+    try:
+        return web.json_response(
+            (await DidIndyManager(context.profile).register(body.get("seed")))
+        )
+    except WalletError as e:
+        raise web.HTTPBadRequest(reason=str(e))
+
+
+async def register(app: web.Application):
+    """Register routes."""
+    app.add_routes([web.post("/did/indy/create", create_indy_did)])
```

Comment:
Would it make sense (and would it be a manageable/reasonable overhead) to have a path-parameter based route to handle creation of different did types? Maybe this is a non-issue, but I think it would be nicer (and easier for users to consume) to be able to get to DID creation by just invoking `/did/{type}/create` rather than having to find the right endpoint to call.

Currently there aren't too many options (but we already have a couple different endpoints to create DIDs), however I wonder whether we're going to end-up in a situation where there is a proliferation of endpoints due to growing number of options in did method types.

Reply:
In some ways that would be nicer. One factor is how consistent the payload options are going to be. Having them separate make it easier to give detailed examples related to that did method. The idea here is there would be multiple endpoints but the did method is somewhat like a path variable. Like if I want to make a did:indy it would be /did/indy/create and if I want a did:web it would be did/web/create and this would be consistent for all did methods supported by core acapy and official plugins.

Reply:
Right. Your point about the consistent payload is valid. Is there a way for us to make sure the payload is consistent for new DID method registrations? I am not too familiar with options other than indy/sov right now.

At the very least we should maybe keep this pattern in mind as we move forward so we can implement it IF we see it makes sense and it is possible (I really am not a big fan of many endpoints that basically do the same thing like we already have for several functions).

Reply:
This idea is from here https://hackmd.io/06YnFvnVQLyP_BzGQjEdIg?view#Registrars but I haven't implemented this exactly this way. The endpoints would have a consistent `features` dictionary with options to support. I'm not sure if the `seed` option would fall under features. I think it's going to be hard to create a simple options payload for all these did methods. That's a main reason why we want to move away from the current `wallet/did/create` endpoint.

Possibly it would be better to get this *interface* narrowed down before adding this endpoint.